modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleLocalServiceImpl.java

```diff
@@ -507,12 +507,34 @@ public KBArticle deleteKBArticle(long resourcePrimKey)
 	public void deleteKBArticles(long groupId, long parentResourcePrimKey)
 		throws PortalException {
 
-		List<KBArticle> childKBArticles = getKBArticles(
-			groupId, parentResourcePrimKey, WorkflowConstants.STATUS_ANY,
-			QueryUtil.ALL_POS, QueryUtil.ALL_POS, null);
+		deleteKBArticles(groupId, parentResourcePrimKey, true);
+	}
+
+	@Override
+	public void deleteKBArticles(
+			long groupId, long parentResourcePrimKey,
+			boolean includeTrashedEntries)
+		throws PortalException {
+
+		List<KBArticle> childKBArticles = new ArrayList<>();
+
+		childKBArticles.addAll(
+			getKBArticles(
+				groupId, parentResourcePrimKey, WorkflowConstants.STATUS_ANY,
+				QueryUtil.ALL_POS, QueryUtil.ALL_POS, null));
+
+		childKBArticles.addAll(
+			getKBArticles(
+				groupId, parentResourcePrimKey,
+				WorkflowConstants.STATUS_IN_TRASH, QueryUtil.ALL_POS,
+				QueryUtil.ALL_POS, null));
 
 		for (KBArticle childKBArticle : childKBArticles) {
-			kbArticleLocalService.deleteKBArticle(childKBArticle);
+			if (includeTrashedEntries ||
+				!_trashHelper.isInTrashExplicitly(childKBArticle)) {
+
+				kbArticleLocalService.deleteKBArticle(childKBArticle);
+			}
```

Comment:
Not for now, as the problem was there already, but we should be using an actionable dynamic query here; loading all group articles into a list could take huge amounts of memory. You can create a ticket for this and fix it later when we got the time.

Reply:
I've created the issue [Use actionable dynamic query to avoid getting all data in memory in KBArticleLocalServiceImpl](https://liferay.atlassian.net/browse/LPS-202953)